csrc/ir_nodes.cpp

```diff
@@ -3210,9 +3208,8 @@ std::vector<Val*> PadOp::getPadWidths() const {
   return {getPadWidthInputBegin(), getPadWidthInputEnd()};
 }
 
-std::pair<Val*, Val*> PadOp::getPadWidths(int axis) const {
-  const auto num_dims =
-      static_cast<int>(out()->as<TensorView>()->getRootDomain().size());
+std::pair<Val*, Val*> PadOp::getPadWidths(int64_t axis) const {
```

Comment:
What does clang-tidy complain about this function?

Reply:
Error (CLANGTIDY) [bugprone-implicit-widening-of-multiplication-result,-warnings-as-errors] performing an implicit widening conversion to type `__gnu_cxx::__normal_iterator<nvfuser::Val *const *, std::vector<nvfuser::Val *>>::difference_type` (aka 'long') of a multiplication performed in type `int`

        3218  |  TORCH_CHECK(axis >= 0 && axis < num_dims, "Invalid axis: ", axis);
        3219  |
        3220  |  return std::make_pair(
    >>> 3221  |      (*(getPadWidthInputBegin() + axis * 2))->as<Val>(),
        3222  |      (*(getPadWidthInputBegin() + axis * 2 + 1))->as<Val>());
        3223  |}
        3224  |

Reply:
Hmm, so it's complaining about `axis * 2`, which is type `int`, being cast to `long`? `getPadWidthInputBegin()` just returns an iterator, so I thought it's quite a common thing to do to increment by some `int` value. For example:

```
std::vector<XYZ> v;
v.push_back(...);
```

Then:

```
for (auto it = v.begin(); it != v.end(); it = it + 1) {
  ...
}
```

This also increments an iterator by an `int` value, which is just 1. Is this also considered a bad pattern?

Reply:
Yes, it does incur the cast. My question is this is so common, but is it still considered a bad practice by clang-tidy?

Reply:
The iterator forces the multiplication to be cast to `long`. It is only when you use multiply the iterator offset will you see this clang-tidy error.

Reply:
Hmm, I don't understand why the multiplication is cast to `long`.

It seems this doesn't incur any cast:
```
int axis = 1;
for (auto it = v.begin(); it != v.end(); it = it + axis) {
  std::cout << *it << std::endl;
}
```

I don't know why this doesn't need a cast as `axis` is also `int` and should have the same type as `axis * 2`.

Reply:
OK, I guess now I see why it's flagged:

https://clang.llvm.org/extra/clang-tidy/checks/bugprone/implicit-widening-of-multiplication-result.html

Reply:
For the `int axis=1` case, the cast is implicit.

MOVSX | r1, r/m | Move with sign extension.

https://godbolt.org/z/j5GYnTWcb
```asm
mov     eax, DWORD PTR [rbp-20]
movsx   rdx, eax
lea     rax, [rbp-88]
mov     rsi, rdx
mov     rdi, rax
call    __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >::operator+(long) const
mov     QWORD PTR [rbp-88], rax
```